test/windows_test.dart

```diff
@@ -29,6 +29,11 @@ void main() {
     expect(context.extension(r'a.b/c'), r'');
     expect(context.extension(r'foo.dart\'), '.dart');
     expect(context.extension(r'foo.dart\\'), '.dart');
+    expect(context.extension('a.b/..', 2), '');
+    expect(context.extension('foo.bar.dart.js', 2), '.dart.js');
+    expect(context.extension(r'foo.bar.dart.js', 3), '.bar.dart.js');
+    expect(context.extension(r'foo.bar.dart.js', 10), '.bar.dart.js');
+    expect(context.extension('a.b/c.d', 2), '.d');
```

Comment:
What is the behavior if you pass `0`? What should the behavior be? Can we document it?

Reply:
I am returning the full path (base path + extension) is `level=0`. Added tests and doc comments.

Reply:
I can't see the reason to replicate the behavior of `baseName` as a special value here. If anything I'd expect it to always return the empty string. It feels strange for the result to get shorter and shorter, then suddenly jump to longer at `0`.

@jakemac53 WDYT?

Reply:
I had the same thought, I think we should just disallow `0`

Reply:
Disallowing 0 now.